src/Services/lessonsPlansScrapper.service.ts

```diff
@@ -0,0 +1,156 @@
+import { MajorLessonsPlanObject } from '../Types/majorLessonsPlanObject.type';
+import { LessonsPlanURLObject } from '../Types/lessonsPlanURLObject.type';
+import { LessonsPlanEntry } from '../Types/lessonsPlanEntry.type';
+import { getURLWithoutQuery } from '../utils/getURLWithoutQuery';
+import { transformCSVHeader } from '../utils/transformCSVHeader';
+import { transformCSVField } from '../utils/transformCSVField';
+import { checkIfFaculty } from '../utils/checkIfFaculty';
+import cheerio from 'cheerio';
+import Papa from 'papaparse';
+import * as _ from 'lodash';
+import axios from 'axios';
+
+const getDataFromCSV = async (
+    url: string,
+    type: 'main' | 'seminar' | 'faculty'
+): Promise<LessonsPlanEntry[]> => {
+    try {
+        const parserConfig = {
+            header: true,
+            dynamicTyping: true,
+            transformHeader: transformCSVHeader,
+            transform: transformCSVField,
+        };
+
+        const csv = await axios.get(
+            `https://inf.ug.edu.pl/plan/${url}&format=csv`
+        );
+
+        const results = Papa.parse(csv.data, parserConfig).data;
+
+        const withoutLastLine = results.splice(0, results.length - 1);
```

Comment:
maybe you could add a comment why you get rid of the last line, it looks weird

Reply:
I've changed the name of a constant and how it is counted, I hope now it's more understandable